src/compiler/crystal/tools/git.cr

```diff
@@ -0,0 +1,23 @@
+module Crystal::Git
+  class_property executable = "git"
+
+  # Tries to run git command with args.
+  # Yields block if exec fails or process status is not success.
+  def self.git_command(args, output : Process::Stdio = Process::Redirect::Close)
+    status = Process.run(executable, args, output: output)
+    yield unless status.success?
+    status
+  rescue IO::Error
+    yield
+  end
+
+  def self.git_capture(args)
+    String.build do |io|
+      git_command(args, output: io) { yield }
+    end
+  end
```

Comment:
@straight-shoota Why `yield` if use can just use nilable return type to signify failure?

Reply:
Oh, I guess that's just left over from an initial implementation. I agree that returning `nil` seems more reasonable.